### PR TITLE
Make post-processing wait until completion, then report stats

### PR DESCRIPTION
**Pull request by lathanh**
## Checklist

Not all may apply

- [x] Code changes (must work)
- [ ] ~~Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)~~
- [ ] ~~Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/coverage.yml` as necessary.~~
- [ ] All other unit and integration tests passing
- [ ] ~~Update validation for project config yaml file changes~~
- [ ] ~~Update existing documentation~~
- [ ] ~~Run a small batch run on Eagle to make sure it all works if you made changes that will affect Eagle~~
- [ ] ~~Add to the changelog_dev.rst file and propose migration text in the pull request~~

### CODE REVIEW

File: buildstockbatch/gcp/gcp.py [hunk 1]
```diff
@@ -884,7 +893,86 @@ def start_combine_results_job_on_cloud(self, results_dir, do_timeseries=True):
...
                 )
-                break
+                return
+
+        # Monitor job/execution status, starting by polling the Job for an Execution
+        logger.info("Waiting for execution to begin...")
+        job_start_time = datetime.now()
+        jobs_client = run_v2.JobsClient()
+        job = jobs_client.get_job(name=self.postprocessing_job_name)
+        while not job.latest_created_execution:
```
Comment (nweires) [inline buildstockbatch/gcp/gcp.py:914]: Could you get the execution from the return value of `run_job` above instead?

File: buildstockbatch/gcp/gcp.py [hunk 2]
```diff
@@ -884,7 +893,86 @@ def start_combine_results_job_on_cloud(self, results_dir, do_timeseries=True):
...
+            f"Execution has started (after {str(execution_start_time - job_start_time)} "
+            "seconds). Waiting for execution to finish..."
+        )
+
+        # Have an execution; poll that for completion
+        fail_message = None
+        with tqdm.tqdm(
+            desc="Waiting for post-processing execution to complete", bar_format="{desc}: {elapsed}{postfix}"
+        ) as pp_tqdm:
+            spinner_states = ["|", "/", "-", "\\"]
```
Comment (nweires) [inline buildstockbatch/gcp/gcp.py:928]: Fancy! 😄

File: buildstockbatch/gcp/gcp.py [hunk 3]
```diff
@@ -884,7 +893,86 @@ def start_combine_results_job_on_cloud(self, results_dir, do_timeseries=True):
...
+                        fail_message = "🟧 Post-processing execution canceled."
+                        break
+
+                spinner_state = (spinner_state + 1) % len(spinner_states)
+                pp_tqdm.set_postfix_str(spinner_states[spinner_state])
+                pp_tqdm.update()
+
+        if fail_message is not None:
+            # if logged within the tqdm block, the message ends up on the same line as the status
+            logger.warning(f"{fail_message} See {self.postprocessing_job_console_url} for more" " information")
```
Comment (nweires) [inline buildstockbatch/gcp/gcp.py]:
```suggestion
            logger.warning(f"{fail_message} See {self.postprocessing_job_console_url} for more information")
```

File: buildstockbatch/gcp/gcp.py [hunk 4]
```diff
@@ -884,7 +893,86 @@ def start_combine_results_job_on_cloud(self, results_dir, do_timeseries=True):
...
+
+        # Format stats for easy copying into a spreadsheet
+        keys, values = [], []
+
+        def append_stat(key, value):
+            nonlocal keys
+            nonlocal values
+            width = max(len(str(key)), len(str(value)))
+            keys.append(str(key).rjust(width))
+            values.append(str(value).rjust(width))
```
Comment (nweires) [inline buildstockbatch/gcp/gcp.py]: We should probably refactor this so we don't need two copies of it.

**Review by nweires — APPROVED** [review]

File: buildstockbatch/gcp/gcp.py (hunk 1 above)
Comment (lathanh) [reply buildstockbatch/gcp/gcp.py:914]: `run_job` (and `create_job`) [return an Operation](https://cloud.google.com/python/docs/reference/run/latest/google.cloud.run_v2.services.jobs.JobsClient#google_cloud_run_v2_services_jobs_JobsClient_run_job), not a Job (which might have an execution in it), from what I can tell.

File: buildstockbatch/gcp/gcp.py (hunk 3 above)
Comment (lathanh) [reply buildstockbatch/gcp/gcp.py]: Turns out this was the precommit formatter (it un-wrapped the lines, not consolidating the strings as it did so).

**Pull request merged**